assets/js/modules/analytics/components/common/ProfileSelect.test.js

```diff
@@ -130,6 +130,56 @@ describe( 'ProfileSelect', () => {
 		expect( apiFetchMock ).not.toHaveBeenCalled();
 	} );
 
+	it( 'should not render if account ID is not set.', async () => {
+		const { container, registry } = render( <ProfileSelect />, {
+			setupRegistry: ( { dispatch } ) => {
+				const accountID = fixtures.accountsPropertiesProfiles.profiles[ 0 ].accountId; // eslint-disable-line sitekit/acronym-case
+				const propertyID = fixtures.accountsPropertiesProfiles.profiles[ 0 ].webPropertyId; // eslint-disable-line sitekit/acronym-case
+
+				dispatch( STORE_NAME ).setSettings( {} );
+
+				dispatch( STORE_NAME ).receiveGetAccounts( fixtures.accountsPropertiesProfiles.accounts );
+				dispatch( STORE_NAME ).finishResolution( 'getAccounts', [] );
+
+				dispatch( STORE_NAME ).receiveGetProperties( fixtures.accountsPropertiesProfiles.properties, { accountID } );
+				dispatch( STORE_NAME ).finishResolution( 'getProperties', [ accountID ] );
+
+				dispatch( STORE_NAME ).receiveGetProfiles( [], { accountID, propertyID } );
+				dispatch( STORE_NAME ).finishResolution( 'getProfiles', [ accountID, propertyID ] );
+			},
+		} );
+
+		expect( container ).toBeEmptyDOMElement();
+
+		await act( () => registry.dispatch( STORE_NAME ).setAccountID( 'pub-12345678' ) );
+	} );
+
+	it( 'should not render if property ID is not set.', async () => {
+		const { container, registry } = render( <ProfileSelect />, {
+			setupRegistry: ( { dispatch } ) => {
+				const accountID = fixtures.accountsPropertiesProfiles.profiles[ 0 ].accountId; // eslint-disable-line sitekit/acronym-case
+				const propertyID = fixtures.accountsPropertiesProfiles.profiles[ 0 ].webPropertyId; // eslint-disable-line sitekit/acronym-case
+
+				dispatch( STORE_NAME ).setSettings( {} );
+				dispatch( STORE_NAME ).setAccountID( accountID );
+
+				dispatch( STORE_NAME ).receiveGetAccounts( fixtures.accountsPropertiesProfiles.accounts );
+				dispatch( STORE_NAME ).finishResolution( 'getAccounts', [] );
+
+				dispatch( STORE_NAME ).receiveGetProperties( fixtures.accountsPropertiesProfiles.properties, { accountID } );
+				dispatch( STORE_NAME ).finishResolution( 'getProperties', [ accountID ] );
+
+				dispatch( STORE_NAME ).receiveGetProfiles( [], { accountID, propertyID } );
+				dispatch( STORE_NAME ).finishResolution( 'getProfiles', [ accountID, propertyID ] );
+			},
+		} );
+
+		expect( container ).toBeEmptyDOMElement();
+
+		// in order to not get act errors need to do something as DOM is empty at the start and end of this test AND wp-data does lots of stuff
+		await act( () => registry.dispatch( STORE_NAME ).setAccountID( 'pub-12345678' ) );
```

Comment:
We should set the `propertyID` instead, because the `accountID` setting is already set in the setup function. Plus, we need to check that the ProfileSelect element becomes visible after doing these changes.